src/activities/menu/ConfigurationItem.qml

```diff
@@ -404,6 +413,56 @@ Item {
                 }
             }
         }
+        
+        Flow {
+            id: backgroundMusicRow
+            width: parent.width
+            spacing: 5 * ApplicationInfo.ratio
+            
+            property bool haveBackgroundMusicResource: false
+
+            function backgroundMusicChanged() {
+                backgroundMusicRow.haveBackgroundMusicResource = DownloadManager.isDataRegistered(
+                            "backgroundMusic-" + ApplicationInfo.CompressedAudio + ".rcc" )
+                         }
+
+
+            Connections {
+                target: DownloadManager
+                onDownloadFinished: backgroundMusicRow.backgroundMusicChanged()
+            }
+            
+            
+            GCText {
+                id: backgroundMusicText
+                text: qsTr("Background music")
+                fontSize: mediumSize
+                wrapMode: Text.WordWrap
+            }
+            
+            Image {
+                id: backgroundMusicImage
+                sourceSize.height: 30 * ApplicationInfo.ratio
+                source: backgroundMusicRow.haveBackgroundMusicResource ? "qrc:/gcompris/src/core/resource/apply.svg" :
+                "qrc:/gcompris/src/core/resource/cancel.svg"
+            }
+            
+            Button {
+                id: backgroundMusicButton
+                height: 30 * ApplicationInfo.ratio
+                visible: ApplicationInfo.isDownloadAllowed
+                text: backgroundMusicRow ? qsTr("Check for updates") :
+                qsTr("Download")
+                style: GCButtonStyle {}
+                onClicked: {
+                if(DownloadManager.downloadResource(DownloadManager.getBackgroundMusicResources())) {
```

Comment:
There is no automatic download for now which is basically start to download when we start GCompris right? Currently the button only does the job. Can it be done that if the music is enabled and there is no rcc then we download the rcc and get it registered else if the rcc is not registered then get it registered? Do you see it this way only?